user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleInformationMain.java

```diff
@@ -474,12 +479,10 @@ private void setBatchAndSampleInformation() {
   }
 
   private boolean noSamplesRegisteredInExperiment(Experiment experiment) {
-    return sampleInformationService.retrieveSamplesForExperiment(experiment.experimentId())
-        .map(Collection::isEmpty)
-        .onError(error -> {
-          throw new ApplicationException("Unexpected response code : " + error);
-        })
-        .getValue();
+    var result = asyncProjectService
+        .getSamples(context.projectId().orElseThrow().value(), experiment.experimentId().value())
+        .collectList().block();
```

Comment:
Instead of waiting for all elements to be there you could only wait for the first element as it is your concern whether the list is empty.
https://projectreactor.io/docs/core/release/reference/apdx-operatorChoice.html#which.blocking
https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#blockFirst-java.time.Duration-

Reply:
i think cheats the way Flux is working: we don't know how many samples will be send through the Flux. So maybe it is better to use a convenience Method in the Service API instead?

Reply:
Why would this cheat the way flux is working?
The flux publishes elements and instead of waiting for the flux to complete, you can simply wait for the first element. No need for the flux to continue publishing afterwards as you do not want to use the result.

Reply:
Of course you could think about adding a method Mono<Boolean> to determine whether there are any samples in an experiment. Depending on the way the flux is created this would minimize the work the system does. But I think both are legitimate ways to handle the situation.